client/post-editor/editor-sharing/publicize-connection.jsx

```diff
@@ -163,6 +167,35 @@ export class EditorSharingPublicizeConnection extends React.Component {
 		);
 	};
 
+	/**
+	 * If a connection needs reauthentication, display a warning linked to Sharing settings page so users can disconnect and reconnect.
+	 *
+	 * @returns {object} Warning about connection.
+	 */
+	renderMustReauthConnection = () => {
+		const { connection, siteSlug } = this.props;
+		if ( ! connection || connection.status !== 'must_reauth' ) {
+			return;
+		}
+
+		return (
+			<Notice
+				isCompact
+				className="editor-sharing__must-reauth-publicize-connection"
+				status="is-warning"
+				showDismiss={ false }
+			>
+				{ this.props.translate(
+					'Your LinkedIn connection needs to be reauthenticated to continue working – head to Sharing to take care of it.'
+				) }
```

Comment:
Again, this seems to be to have service-generic logic with service-specific messaging. Is that a problem?

Reply:
I originally phrased this as `Your connection needs...` but editorial added the `LinkedIn`.
Is not an issue right now because we're only using this for LinkedIn and has specific stuff in place in wpcom. If we ever need it for something else we can differentiate by service and add a generic name.